rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,124 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator and its companion toolset generated from the Nix source tree are built using the typical `./configure` shell script that relies on autoconf and the standard GNU Make utility.
+
+This build system became clunky and plastered, and consequently hard to understand, modify, improve and port to other systems besides Linux.
+
+Such state of things hinders development, specially outside the Linux and NixOS niches.
+
+In light of this, we propose a novel, from-scratch alternative build infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring;
+- improved cross-platform support, especially in other programming environments, including but not limited to Unix-like operating systems;
+- shorter build times;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of files written in Meson should be included in the Nix repository, in order to describe how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files etc.)
+
+This novel build infrastructure should be able to provide at least all the features already present on the current quasi-autotools implementation, possibly with a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Some typical expected interactions are:
+
+- Configuring: `meson setup build_dir`
+- Building: `ninja -C build_dir` (or the backend-agnostic `meson -C build_dir compile`)
+- Installing: `ninja -C build_dir install` (or the backend-agnostic `meson -C build_dir install`)
+
+Usually, commandline options assume the format `-Dname=value`. This small change in user interface when compared to the typical `--name=value` is mostly harmless.
```

Comment:
consider adding:

"A bootstrap shell script has been provided that will configure and setup the build environment for the user"

Reply:
What such script will do, exactly?

Reply:
it can just run the meson commands for the user, the same as the current one :)

Reply:
It looks as useless as creating a configure; make; make install file.

Reply:
for the most part yes, but for some beginners it can be helpful. the current build already has a bootstrap file along the same lines

Reply:
I rather just do instructions too.

You can always just run `genericBuild` or whatever in a `nix-shell` to go through the default build steps. When those don't work, it is better to learn a few tiny details of the meson and ninja CLIs.

Reply:
>  the current build already has a bootstrap file along the same lines

You say this? https://github.com/NixOS/nix/blob/master/bootstrap.sh

This is just because no one ever touches an autotool in real life :) 
Indeed Nixpkgs has an `autoreconfHook` just for cases like that.

In real life situations like downloading a tarball from Sourceforge, it is previously cleaned up in order to remove these autotool files and leave only a cute `./configure` script.

In Meson, such a bootstrap file is unneeded.

Reply:
That appears to be a completely cargo-culted `autogen.sh` with a different name... the only line that meaningfully does anything is the last one anyway.